src/mca/plm/slurm/plm_slurm_component.c

```diff
@@ -117,40 +105,44 @@ static int plm_slurm_open(void)
 
 static int prte_mca_plm_slurm_component_query(pmix_mca_base_module_t **module, int *priority)
 {
-    const pmix_mca_base_var_t *var;
-    pmix_status_t rc;
+    FILE *fp;
+    char version[1024], *ptr;
+    int major, minor;
 
     /* Are we running under a SLURM job? */
-
     if (NULL != getenv("SLURM_JOBID")) {
         *priority = 75;
 
         PMIX_OUTPUT_VERBOSE((1, prte_plm_base_framework.framework_output,
                              "%s plm:slurm: available for selection",
                              PRTE_NAME_PRINT(PRTE_PROC_MY_NAME)));
 
-    prte_mca_plm_slurm_component.custom_args = NULL;
-
-        // if we were are warning about externally set custom args, then
-        // check to see if that was done
-        if (!prte_mca_plm_slurm_component.slurm_warning_msg &&
-            NULL == force_args) {
-            // check for custom args
-            rc = pmix_mca_base_var_get(prte_mca_plm_slurm_component.custom_args_index, &var);
-            if (PMIX_SUCCESS == rc) {
-                // the variable was set - see who set it
-                if (PMIX_MCA_BASE_VAR_SOURCE_ENV == var->mbv_source) {
-                    // set in the environment - warn
-                    pmix_show_help("help-plm-slurm.txt", "custom-args-in-env", true,
-                                   custom_args);
-                }
-            }
+        // check the version
+        fp = popen("srun --version", "r");
+        if (NULL == fp) {
+            // cannot run srun, so we cannot support this job
+            *module = NULL;
+            return PRTE_ERROR;
         }
-
-        if (NULL != force_args) {
-            prte_mca_plm_slurm_component.custom_args = force_args;
-        } else if (NULL != custom_args) {
-            prte_mca_plm_slurm_component.custom_args = custom_args;
+        if (NULL == fgets(version, sizeof(version), fp)) {
+            pclose(fp);
+            *module = NULL;
+            return PRTE_ERROR;
+        }
+        pclose(fp);
+        // parse on the dots
+        major = strtol(&version[6], &ptr, 10);
+        ++ptr;
+        minor = strtol(ptr, NULL, 10);
+
+        if (23 > major) {
+            prte_mca_plm_slurm_component.early = true;
+        } else if (23 < major) {
+            prte_mca_plm_slurm_component.early = false;
+        } else if (11 < minor) {
```

Comment:
`11 > minor`

Reply:
I believe the current comparison is actually correct. If we are in the major=23 series, then any minor release less than 11 (i.e., earlier than Nov of that year) is an "early" release that does not use the `--external-launcher` option.

Reply:
Oh - yeah, I do have it backwards! Thx!